File: src/test/java/sirius/db/text/ReduceCharactersProcessorTest.java
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text;

import org.junit.Test;

public class ReduceCharactersProcessorTest extends TokenProcessorTest {

    @Test
    public void testTokenizing() {
        assertExactTokenizing("Hello", new ReduceCharacterProcessor(), "Hello");
        assertExactTokenizing("Höllo", new ReduceCharacterProcessor(), "Hoello");
        assertExactTokenizing("ÄHölloÜ", new ReduceCharacterProcessor(), "AeHoelloUe");
        assertExactTokenizing("ÄÖßÜ", new ReduceCharacterProcessor(), "AeOessUe");
        assertExactTokenizing("Hüüllüü", new ReduceCharacterProcessor(), "Hueuellueue");
        assertExactTokenizing(" Hüü llüü ", new ReduceCharacterProcessor(), " Hueue llueue ");
    }
}

File: src/test/kotlin/sirius/db/text/ReduceCharactersProcessorTest.kt
/*
 * Made with all the love in the world
 * by scireum in Remshalden, Germany
 *
 * Copyright by scireum GmbH
 * http://www.scireum.de - info@scireum.de
 */

package sirius.db.text

import org.junit.jupiter.api.Test

/**
 * Tests the [ReduceCharacterProcessor].
 */
class ReduceCharactersProcessorTest : TokenProcessorTest() {
    @Test
    fun tokenizing() {
        assertExactTokenizing("Hello", ReduceCharacterProcessor(), "Hello")
        assertExactTokenizing("Höllo", ReduceCharacterProcessor(), "Hoello")
        assertExactTokenizing("ÄHölloÜ", ReduceCharacterProcessor(), "AEHoelloUE")
        assertExactTokenizing("ÄÖßÜ", ReduceCharacterProcessor(), "AEOEssUE")
        assertExactTokenizing("Hüüllüü", ReduceCharacterProcessor(), "Hueuellueue")
        assertExactTokenizing(" Hüü llüü ", ReduceCharacterProcessor(), " Hueue llueue ")
    }
}
